### PR TITLE
fix for new rewrite test case and add table test to replacer

### PR DESCRIPTION
**Pull request by pedronasser**

### CODE REVIEW

File: caddyhttp/rewrite/to_test.go [hunk 1]
```diff
@@ -22,7 +22,8 @@ func TestTo(t *testing.T) {
 		{"/?a=b", "/testfile /index.php?{query}", "/testfile?a=b"},
 		{"/?a=b", "/testdir /index.php?{query}", "/index.php?a=b"},
 		{"/?a=b", "/testdir/ /index.php?{query}", "/testdir/?a=b"},
-		{"/test?url=http://caddyserver.com", " /p/{path}?{query}", "/p/test?url=http://caddyserver.com"},
```
Comment (abiosoft) [inline caddyhttp/rewrite/to_test.go]: any reason why you removed this test ?

File: caddyhttp/rewrite/to.go [hunk 2]
```diff
@@ -28,6 +28,10 @@ func To(fs http.FileSystem, r *http.Request, to string, replacer httpserver.Repl
 			query = tparts[1]
 		}
 
+		if strings.HasSuffix(tparts[0], "/") && !strings.HasSuffix(t, "/") {
+			t += "/"
+		}
+
```
Comment (abiosoft) [inline caddyhttp/rewrite/to.go]: With this, I do not think the similar check below is needed anymore. Maybe move the `// add trailing slash for directories, if present` comments over here and delete the other one.

File: caddyhttp/rewrite/to_test.go (hunk 1 above)
Comment (pedronasser) [inline caddyhttp/rewrite/to_test.go]: Not removed, just replaced.
No need for the `caddyserver.com`. The problem is in the `http://`

File: caddyhttp/rewrite/to.go (hunk 2 above)
Comment (pedronasser) [inline caddyhttp/rewrite/to.go]: I think they are different cases, but I will work on it

File: caddyhttp/rewrite/to_test.go (hunk 1 above)
Comment (abiosoft) [inline caddyhttp/rewrite/to_test.go]: For readability since the emphasis is on trailing slash, there can be additional test for `http://caddyserver.com/`

Comment (pedronasser) [inline caddyhttp/rewrite/to_test.go]: The enphasis on that test was on the double slashes. But we could keep the full address as well

File: caddyhttp/rewrite/to.go (hunk 2 above)
Comment (mholt) [inline caddyhttp/rewrite/to.go]: Finally getting around to this PR - what have you decided about this?

**Comment by mholt** [comment]
Thanks @pedronasser! :+1:

**Pull request merged**